toolkit/tools/pkg/licensecheck/licensecheck.go

```diff
@@ -0,0 +1,411 @@
+		}
+		l.simpleToolChroot = nil
+	}
+	return nil
+}
+
+// CheckLicenses will scan all .rpm files in the chroot for bad licenses. The results can be accessed with FormatResults() or GetAllResults().
+func (l *LicenseChecker) CheckLicenses() error {
+	if l.simpleToolChroot == nil {
+		return fmt.Errorf("chroot has not been initialized")
+	}
+
+	l.results = []LicenseCheckResult{}
+
+	err := l.simpleToolChroot.RunInChroot(func() (searchErr error) {
+		l.results, searchErr = l.runLicenseCheckInChroot()
+		return searchErr
+	})
+	if err != nil {
+		return fmt.Errorf("failed to scan for license issues:\n%w", err)
+	}
+
+	for _, result := range l.results {
+		if result.err != nil {
+			logger.Log.Errorf("Failed to scan (%s) for license issues: %v", result.RpmPath, result.err)
+		}
+	}
+
+	// Sort the results by RPM path
+	// This is done to ensure that the output is deterministic
+	sort.Slice(l.results, func(i, j int) bool {
+		return l.results[i].RpmPath < l.results[j].RpmPath
+	})
+	return nil
+}
+
+// GetAllResults returns the results of the search, split into:
+// - All results: Every scan result
+// - Any result that has at least one warning
+// - Any result that has at least one error
+// There may be overlap between the lists. If pedantic is true, warnings will be treated as errors.
+func (l *LicenseChecker) GetAllResults(pedantic bool) (all, warnings, errors []LicenseCheckResult) {
+	all = []LicenseCheckResult{}
+	warnings = []LicenseCheckResult{}
+	errors = []LicenseCheckResult{}
+	for _, result := range l.results {
+		all = append(all, result)
+		if result.HasBadResult() || (pedantic && result.HasWarningResult()) {
+			errors = append(errors, result)
+		}
+		if result.HasWarningResult() && !pedantic {
+			warnings = append(warnings, result)
+		}
+	}
+	return all, warnings, errors
+}
+
+// FormatResults formats the results of the search to a string. Results will be ordered as follows:
+// - Packages with warnings only, sorted alphabetically
+// - Packages with errors (and possibly warnings), sorted alphabetically
+// If pedantic is true, warnings will be treated as errors.
+func (l *LicenseChecker) FormatResults(pedantic bool) string {
+	var sb strings.Builder
+	results := sortAndFilterResults(l.results)
+
+	// Print warnings first, but only if they don't also have an error
+	if !pedantic {
+		for _, result := range results {
+			if result.HasWarningResult() && !result.HasBadResult() {
+				sb.WriteString(printWarning(result))
+			}
+		}
+	}
+
+	// Now print the errors
+	for _, result := range results {
+		if result.HasBadResult() || (pedantic && result.HasWarningResult()) {
+			sb.WriteString(printError(result, pedantic))
+			if !pedantic && result.HasWarningResult() {
+				// If pedantic was set, the warning was already printed as an error.
+				// Otherwise print the warning now.
+				sb.WriteString(printWarning(result))
+			}
+		}
+	}
+
+	return sb.String()
+}
+
+func printWarning(result LicenseCheckResult) string {
```

Comment:
Naming may be a bit confusing: we're not printing anything in here and in `printError`. How about `formatWarning` or `buildWarning`?